solidity/contracts/Chainlinked.sol

```diff
@@ -24,31 +24,21 @@ contract Chainlinked {
     string _callbackFunctionSignature
   ) internal pure returns (ChainlinkLib.Run memory) {
     ChainlinkLib.Run memory run;
-    Buffer.init(run.buf, 128);
-    run.jobId = _jobId;
-    run.callbackAddress = _callbackAddress;
-    run.callbackFunctionId = bytes4(keccak256(_callbackFunctionSignature));
-    run.buf.startMap();
-
-    return run;
+    return run.initialize(_jobId, _callbackAddress, _callbackFunctionSignature);
   }
 
   function chainlinkRequest(ChainlinkLib.Run memory _run, uint256 _wei)
     internal
     returns(bytes32)
   {
-    bytes32 requestId = keccak256(this, requests++);
-    bytes memory requestDataABI = abi.encodeWithSelector(
-      oracleFid,
-      clArgsVersion,
-      _run.jobId,
-      _run.callbackAddress,
-      _run.callbackFunctionId,
-      requestId,
-      _run.close());
-    require(link.transferAndCall(oracle, _wei, requestDataABI));
-
-    return requestId;
+    _run.externalId = keccak256(this, requests++);
+    _run.close();
+    require(link.transferAndCall(
+      oracle,
+      _wei,
+      _run.encodeForOracle(oracleFid, clArgsVersion)));
+
+    return _run.externalId;
```

Comment:
If you run two `chainlinkRequest`s, you will need two `Run`s otherwise you will invalidate the first. `close()` will also give an issue. I bring this up because it illustrates how `Run` is stateful and sensitive to the order of calls.

Reply:
We would still be in a world of hurt if they tried to add after. It's fine that we don't allow that but it should be communicated, perhaps in the method name. I'll ping you offline.